### ISSUE TITLE
Android crash - Error while updating property 'streamId' of a view managed by OTSubscriberViewManager

### ISSUE THREAD

**Issue by deepakverma4** [opentok/opentok-react-native#462]
![Screenshot_2020-12-19-13-57-21-00_5db42c1825fa5eb6b46927e4464f44b6](https://user-images.githubusercontent.com/31440793/102684948-ebe5b600-4202-11eb-83c5-16a7e966627b.jpg)

I'm facing this crash randomly when a user who shares screen leaves meeting.  Any idea why is it happening?

**Comment by deepakverma4** [comment]
@msach22 Bro any idea why is it happening?

**Comment by dl748** [comment]
I'm having the same error in a different section of code.



```
java.lang.NullPointerException: Attempt to invoke virtual method 'int java.lang.Object.hashCode()' on a null object reference

java.util.concurrent.ConcurrentHashMap.get ConcurrentHashMap.java:944
com.opentokreactnative.OTPublisherLayout.createPublisherView OTPublisherLayout.java:36
com.opentokreactnative.OTPublisherViewManager.setPublisherId OTPublisherViewManager.java:31
java.lang.reflect.Method.invoke Method.java
```

Looks like mPublisher is NOT null. and getSession() is returning a session, but getSessionId() is returning null.  I'm wondering if there is some disconnection after the publish but before the render.

**Comment by it-bp-msc** [comment]
@dl748, have you found any ways to deal with the issue you encountered? Thanks

**Comment by dl748** [comment]
@it-bp-msc we narrowed it down to a permission check on android. If we let OpenTok do the permission check, it executed "twice", once for video, once for audio (each box). When we asked for permission BEFORE we instantiated OpenTok (aka we did it ourselves), the error went away.

**Comment by it-bp-msc** [comment]
> @it-bp-msc we narrowed it down to a permission check on android. If we let OpenTok do the permission check, it executed "twice", once for video, once for audio (each box). When we asked for permission BEFORE we instantiated OpenTok (aka we did it ourselves), the error went away.

@dl748 , could you please tell me where exactly did you ask for the permissions? Is it somewhere in the native part or inside the JS module? Thanks

**Comment by it-bp-msc** [comment]
@dl748 , thanks for you insights 👍🏻

**Comment by nicholasio** [comment]
We were able to fix this issue with this fix: https://github.com/10up/opentok-react-native/pull/2/files

**Comment by abdulajet** [comment]
Would someone be up for PRing these changes into main? cc @nicholasio

**Comment by enricop89** [comment]
This should be fixed with `0.18.0`

**Comment by vishalcodiant** [comment]
fixed https://github.com/opentok/opentok-react-native/pull/525
